workflow/controller/operator.go

```diff
@@ -2720,6 +2776,7 @@ func (woc *wfOperationCtx) loadExecutionSpec() (wfv1.TemplateReferenceHolder, wf
 
 	executionParameters := woc.wf.Spec.Arguments
 
+	woc.wfSpec = &woc.wf.Spec
```

Comment:
Why is this necessary if we already do this https://github.com/argoproj/argo/blob/48d86f03ce9ac49f5099c46dd7995305eb59eb11/workflow/controller/operator.go#L121?